Makefile

```diff
@@ -73,7 +73,7 @@ publish: ## Generate release manifests and publish a versioned container image.
 helm:  ## Generate Helm Chart
 	cd charts;helm lint karpenter;helm package karpenter;helm repo index .
 
-docs: ## Generate Docs
+docs: toolchain ## Generate Docs
```

Comment:
I'm wary of adding toolchain into the critical path of any of these commands. Toolchain should be run once by developers when starting development on the project. I see that you've made the kubebuilder bit idempotent, but the go install is a fairly expensive call. 

My preference would be to leave the current dev flow as is.

Reply:
I can add a bit more information to the target so it's more clear why the target is failing if someone hasn't run the toolchain target yet..

Reply:
Awesome thanks